### PR TITLE
ci(riot): run tests with pytest latest

### PR DESCRIPTION
**Pull request by mabdinur**
## Description

Run ddtrace test suite with pytest latest

### CODE REVIEW

File: riotfile.py [hunk 1]
```diff
@@ -102,7 +102,7 @@ def select_pys(min_version=MIN_PYTHON_VERSION, max_version=MAX_PYTHON_VERSION):
 venv = Venv(
     pkgs={
         "mock": latest,
-        "pytest": "<7.0.0",
+        **({"pytest": "<7.0.0"} if sys.version_info < (3, 11) else {"pytest": "==7.2.0"}),
```
Comment (P403n1x87) [inline riotfile.py:105]: I think we normally create subenvs to implement this kind of logic

File: pytest.ini [hunk 2]
```diff
@@ -0,0 +1,2 @@
+[pytest]
+ddtrace = 1
```
Comment (mabdinur) [inline pytest.ini:2]:
```suggestion
ddtrace = 1

```

**Comment by mabdinur** [comment]
Enabling ci visibility will trace the execution of tests. Tests that use the global tracer to generate traces might be impacted.

File: setup.cfg [hunk 3]
```diff
@@ -33,6 +33,7 @@ rst-directives = py:data
 # DEV: We use `conftest.py` as a local pytest plugin to configure hooks for collection
 [tool:pytest]
 # --cov-report is intentionally empty else pytest-cov will default to generating a report
+ddtrace = 1
```
Comment (mabdinur) [inline setup.cfg:36]:
```suggestion
```

**Comment by ZStriker19** [comment]
Could you add any further info as to why we weren't doing this before?

**Comment by mabdinur** [comment]
> Could you add any further info as to why we weren't doing this before?

We pinned pytest in the follow [PR](https://github.com/DataDog/dd-trace-py/pull/3227) to resolve a [pytest-importlib incompatibility](https://github.com/pytest-dev/pytest/issues/9608).

pytest==7.0.1 resolved this issue

**Review by emmettbutler — APPROVED** [review]

**Review by brettlangdon — APPROVED** [review]

**Pull request merged**